### PR TITLE
Revise VaultServer api

### PR DESCRIPTION
**Pull request by tamalsaha**
Signed-off-by: Tamal Saha <tamal@appscode.com>

### CODE REVIEW

File: apis/kubevault/v1alpha1/types.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	// Rejects attempt to delete database using ValidationWebhook.
+	TerminationPolicyDoNotTerminate TerminationPolicy = "DoNotTerminate"
+)
+
+// +kubebuilder:validation:Enum=primary;standby;stats
+type ServiceAlias string
+
+const (
+	PrimaryServiceAlias ServiceAlias = "primary"
+	StandbyServiceAlias ServiceAlias = "standby"
```
Comment (tamalsaha) [inline apis/kubevault/v1alpha1/types.go:60]: Vault server service

File: apis/kubevault/v1alpha1/types.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import (
+	ofst "kmodules.xyz/offshoot-api/api/v1"
+)
+
+// +kubebuilder:validation:Enum=Provisioning;DataRestoring;Ready;Critical;NotReady;Halted
+type DatabasePhase string
+
+const (
+	// used for Databases that are currently provisioning
+	DatabasePhaseProvisioning DatabasePhase = "Provisioning"
```
Comment (sakiib) [inline apis/kubevault/v1alpha1/types.go:28]: VaultServerPhase

File: apis/kubevault/v1alpha1/types.go (hunk 1 above)
Comment (sakiib) [inline apis/kubevault/v1alpha1/types.go:60]: standby--> vault

File: apis/kubevault/v1alpha1/vaultserver_types.go [hunk 3]
```diff
@@ -52,21 +51,18 @@ type VaultServer struct {
 }
 
 type VaultServerSpec struct {
-	// Number of replicas to deploy for a Vault deployment.
-	// If unspecified, defaults to 1.
-	// +optional
-	Replicas *int32 `json:"replicas,omitempty" protobuf:"varint,1,opt,name=replicas"`
+	// Version of MongoDB to be deployed.
```
Comment (sakiib) [inline apis/kubevault/v1alpha1/vaultserver_types.go:54]: MongoDB --> VaultServer

**Pull request closed without merging**